### PR TITLE
Update node-coap & types/readable-stream dependency and recreate package-lock.json

### PR DESCRIPTION
**Pull request by danielpeintner**
fixes https://github.com/eclipse-thingweb/node-wot/issues/1292

### CODE REVIEW

**Comment by danielpeintner** [comment]
Note: `npm run build` still causes issues...


```
node_modules/coap/node_modules/@types/readable-stream/index.d.ts:355:9 - error TS2416: Property 'pipe' in type 'Duplex' is not assignable to the same property in base type '_Writable'.
  Type '<S extends _IWritable>(dest: S, pipeOpts?: { end?: boolean | undefined; } | undefined) => S' is not assignable to type '<T extends WritableStream>(destination: T, options?: { end?: boolean | undefined; } | undefined) => T'.
    Types of parameters 'dest' and 'destination' are incompatible.
      Type 'T' is not assignable to type '_IWritable'.
        Type 'WritableStream' is not assignable to type '_IWritable'.
          The types returned by 'end(...)' are incompatible between these types.
            Type 'void' is not assignable to type '_IWritable'.

355         pipe<S extends _IWritable>(dest: S, pipeOpts?: { end?: boolean | undefined }): S;
            ~~~~

node_modules/mqtt/node_modules/@types/readable-stream/index.d.ts:355:9 - error TS2416: Property 'pipe' in type 'Duplex' is not assignable to the same property in base type '_Writable'.
  Type '<S extends _IWritable>(dest: S, pipeOpts?: { end?: boolean | undefined; } | undefined) => S' is not assignable to type '<T extends WritableStream>(destination: T, options?: { end?: boolean | undefined; } | undefined) => T'.
    Types of parameters 'dest' and 'destination' are incompatible.
      Type 'T' is not assignable to type '_IWritable'.
        Type 'WritableStream' is not assignable to type '_IWritable'.
          The types returned by 'end(...)' are incompatible between these types.
            Type 'void' is not assignable to type '_IWritable'.

355         pipe<S extends _IWritable>(dest: S, pipeOpts?: { end?: boolean | undefined }): S;
            ~~~~

packages/examples/src/bindings/coap/example-client.ts:17:35 - error TS2307: Cannot find module '@node-wot/binding-coap' or its corresponding type declarations.

17 import { CoapClientFactory } from "@node-wot/binding-coap";
                                     ~~~~~~~~~~~~~~~~~~~~~~~~

packages/examples/src/bindings/coap/example-server.ts:17:28 - error TS2307: Cannot find module '@node-wot/binding-coap' or its corresponding type declarations.

17 import { CoapServer } from "@node-wot/binding-coap";
                              ~~~~~~~~~~~~~~~~~~~~~~~~

packages/examples/src/quickstart/presence-sensor.ts:20:34 - error TS2307: Cannot find module '@node-wot/binding-mqtt' or its corresponding type declarations.

20 import { MqttBrokerServer } from "@node-wot/binding-mqtt";
                                    ~~~~~~~~~~~~~~~~~~~~~~~~

packages/examples/src/quickstart/smart-clock.ts:19:28 - error TS2307: Cannot find module '@node-wot/binding-coap' or its corresponding type declarations.

19 import { CoapServer } from "@node-wot/binding-coap";
                              ~~~~~~~~~~~~~~~~~~~~~~~~


Found 6 errors.
```

**Comment by danielpeintner** [comment]
Mhh, the latest `package-lock.json` file has @types/readable-stream in v4 only.
Still it is not working... 🤷‍♂️

**Comment by danielpeintner** [comment]
> this seems to actually be an upstream issue with the type definitions themselves

Can you point me to what you mean by that?

**Comment by JKRhb** [comment]
> > this seems to actually be an upstream issue with the type definitions themselves
> 
> Can you point me to what you mean by that?

I got the impression that the type definitions themselves are invalid, since VSCode also shows the error for the `pipe` method within the definition of the `Duplex` class when navigating to the respective file in the `node_modules` folder. So maybe this needs to be fixed upstream, in the code of `@types/readable-stream` itself (and is not related to the handling of the dependency in one of our direct dependencies). Although I am still wondering why this error only occurs here and not in `node-coap` :/

**Comment by JKRhb** [comment]
Two months ago, these changes here have been made that affect the `pipe` method in question: https://github.com/DefinitelyTyped/DefinitelyTyped/commit/14677d368cba8c48e55a77df9d7ba8c6dba738b3

**Comment by JKRhb** [comment]
Okay, so if you change the line 


```ts
        pipe<S extends _IWritable>(dest: S, pipeOpts?: { end?: boolean | undefined }): S;
```

in the `Duplex` class to

```ts
        pipe<S extends NodeJS.WritableStream>(dest: S, pipeOpts?: { end?: boolean | undefined }): S;
```

the transpilation works again :) Maybe this has been an oversight when they made the changes linked above.

**Comment by JKRhb** [comment]
I've opened https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69974 as a potential fix for the issue.

**Comment by JKRhb** [comment]
With https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69974 merged, this hopefully should work now 🤞

File: package.json [hunk 1]
```diff
@@ -50,7 +50,7 @@
         "@types/chai-spies": "^1.0.4",
         "@types/mocha": "^9.0.0",
         "@types/node": "16.18.35",
-        "@types/readable-stream": "^2.3.13",
+        "@types/readable-stream": "^4.0.14",
```
Comment (JKRhb) [inline package.json:53]:
```suggestion
        "@types/readable-stream": "^4.0.15",
```

File: package-lock.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
                 "@types/chai-spies": "^1.0.4",
                 "@types/mocha": "^9.0.0",
                 "@types/node": "16.18.35",
-                "@types/readable-stream": "^2.3.13",
+                "@types/readable-stream": "^4.0.14",
```
Comment (JKRhb) [inline package-lock.json:24]:
```suggestion
                "@types/readable-stream": "^4.0.15",
```

File: package-lock.json [hunk 3]
```diff
@@ -1272,9 +1271,9 @@
             "dev": true
         },
         "node_modules/@types/readable-stream": {
-            "version": "2.3.15",
-            "resolved": "https://registry.npmjs.org/@types/readable-stream/-/readable-stream-2.3.15.tgz",
-            "integrity": "sha512-oM5JSKQCcICF1wvGgmecmHldZ48OZamtMxcGGVICOJA8o8cahXC1zEVAif8iwoc5j8etxFaRFnf095+CDsuoFQ==",
+            "version": "4.0.14",
+            "resolved": "https://registry.npmjs.org/@types/readable-stream/-/readable-stream-4.0.14.tgz",
+            "integrity": "sha512-xZn/AuUbCMShGsqH/ehZtGDwQtbx00M9rZ2ENLe4tOjFZ/JFeWMhEZkk2fEe1jAUqqEAURIkFJ7Az/go8mM1/w==",
```
Comment (JKRhb) [inline package-lock.json:1276]:
```suggestion
            "version": "4.0.15",
            "resolved": "https://registry.npmjs.org/@types/readable-stream/-/readable-stream-4.0.15.tgz",
            "integrity": "sha512-oAZ3kw+kJFkEqyh7xORZOku1YAKvsFTogRY8kVl4vHpEKiDkfnSA/My8haRE7fvmix5Zyy+1pwzOi7yycGLBJw==",
            "license": "MIT",
```

**Review by JKRhb — COMMENTED** [review]
With these changes, at least the build should now succeed again :)

**Comment by danielpeintner** [comment]
Thank you very much @JKRhb 👍

It works now as expected and we can merge the PR as is I think.
@relu91 ?

**Comment by JKRhb** [comment]
Hmm, for me the regeneration of the `package-lock.json` yields a somewhat different result:

<details>
<summary>Diff</summary>


  
```diff
diff --git a/package-lock.json b/package-lock.json
index 317224a9..0a01a0b1 100644
--- a/package-lock.json
+++ b/package-lock.json
@@ -1830,53 +1830,6 @@
                 "node": ">= 8"
             }
         },
-        "node_modules/aproba": {
-            "version": "1.2.0",
-            "resolved": "https://registry.npmjs.org/aproba/-/aproba-1.2.0.tgz",
-            "integrity": "sha512-Y9J6ZjXtoYh8RnXVCMOU/ttDmk1aBjunq9vO0ta5x85WDQiQfUF9sIPBITdbiiIVcBo03Hi3jMxigBtsddlXRw==",
-            "optional": true
-        },
-        "node_modules/are-we-there-yet": {
-            "version": "1.1.7",
-            "resolved": "https://registry.npmjs.org/are-we-there-yet/-/are-we-there-yet-1.1.7.tgz",
-            "integrity": "sha512-nxwy40TuMiUGqMyRHgCSWZ9FM4VAoRP4xUYSTv5ImRog+h9yISPbVH7H8fASCIzYn9wlEv4zvFL7uKDMCFQm3g==",
-            "deprecated": "This package is no longer supported.",
-            "optional": true,
-            "dependencies": {
-                "delegates": "^1.0.0",
-                "readable-stream": "^2.0.6"
-            }
-        },
-        "node_modules/are-we-there-yet/node_modules/isarray": {
-            "version": "1.0.0",
-            "resolved": "https://registry.npmjs.org/isarray/-/isarray-1.0.0.tgz",
-            "integrity": "sha512-VLghIWNM6ELQzo7zwmcg0NmTVyWKYjvIeM83yjp0wRDTmUnrM678fQbcKBo6n2CJEF0szoG//ytg+TKla89ALQ==",
-            "optional": true
-        },
-        "node_modules/are-we-there-yet/node_modules/readable-stream": {
-            "version": "2.3.8",
-            "resolved": "https://registry.npmjs.org/readable-stream/-/readable-stream-2.3.8.tgz",
-            "integrity": "sha512-8p0AUk4XODgIewSi0l8Epjs+EVnWiK7NoDIEGU0HhE7+ZyY8D1IMY7odu5lRrFXGg71L15KG8QrPmum45RTtdA==",
-            "optional": true,
-            "dependencies": {
-                "core-util-is": "~1.0.0",
-                "inherits": "~2.0.3",
-                "isarray": "~1.0.0",
-                "process-nextick-args": "~2.0.0",
-                "safe-buffer": "~5.1.1",
-                "string_decoder": "~1.1.1",
-                "util-deprecate": "~1.0.1"
-            }
-        },
-        "node_modules/are-we-there-yet/node_modules/string_decoder": {
-            "version": "1.1.1",
-            "resolved": "https://registry.npmjs.org/string_decoder/-/string_decoder-1.1.1.tgz",
-            "integrity": "sha512-n/ShnvDi6FHbbVfviro+WojiFzv+s8MPMHBczVePfUpDJLwoLT0ht1l4YwBCbi8pJAveEEdnkHyPyTP/mzRfwg==",
-            "optional": true,
-            "dependencies": {
-                "safe-buffer": "~5.1.0"
-            }
-        },
         "node_modules/arg": {
             "version": "4.1.3",
             "resolved": "https://registry.npmjs.org/arg/-/arg-4.1.3.tgz",
@@ -3160,15 +3113,6 @@
                 "node": "^12.22.0 || ^14.17.0 || >=16.0.0"
             }
         },
-        "node_modules/code-point-at": {
-            "version": "1.1.0",
-            "resolved": "https://registry.npmjs.org/code-point-at/-/code-point-at-1.1.0.tgz",
-            "integrity": "sha512-RpAVKQA5T63xEj6/giIbUEtZwJ4UFIc3ZtvEkiaUERylqe8xb5IvqcgOurZLahv93CLKfxcw5YI+DZcUBRyLXA==",
-            "optional": true,
-            "engines": {
-                "node": ">=0.10.0"
-            }
-        },
         "node_modules/color-convert": {
             "version": "2.0.1",
             "resolved": "https://registry.npmjs.org/color-convert/-/color-convert-2.0.1.tgz",
@@ -3307,12 +3251,6 @@
             "integrity": "sha512-ZMkYO/LkF17QvCPqM0gxw8yUzigAOZOSWSHg91FH6orS7vcEj5dVZTidN2fQ14yBSdg97RqhSNwLUXInd52OTA==",
             "dev": true
         },
-        "node_modules/console-control-strings": {
-            "version": "1.1.0",
-            "resolved": "https://registry.npmjs.org/console-control-strings/-/console-control-strings-1.1.0.tgz",
-            "integrity": "sha512-ty/fTekppD2fIwRvnZAVdeOiGd1c7YXEixbgJTNzqcxJWKQnjJ/V1bNEEE6hygpM3WjwHFUVK6HTjWSzV4a8sQ==",
-            "optional": true
-        },
         "node_modules/constants-browserify": {
             "version": "1.0.0",
             "resolved": "https://registry.npmjs.org/constants-browserify/-/constants-browserify-1.0.0.tgz",
@@ -3574,18 +3512,6 @@
                 "node": ">=0.10"
             }
         },
-        "node_modules/decompress-response": {
-            "version": "4.2.1",
-            "resolved": "https://registry.npmjs.org/decompress-response/-/decompress-response-4.2.1.tgz",
-            "integrity": "sha512-jOSne2qbyE+/r8G1VU+G/82LBs2Fs4LAsTiLSHOCOMZQl2OKZ6i8i4IyHemTe+/yIXOtTcRQMzPcgyhoFlqPkw==",
-            "optional": true,
-            "dependencies": {
-                "mimic-response": "^2.0.0"
-            },
-            "engines": {
-                "node": ">=8"
-            }
-        },
         "node_modules/deep-eql": {
             "version": "4.1.4",
             "resolved": "https://registry.npmjs.org/deep-eql/-/deep-eql-4.1.4.tgz",
@@ -3662,12 +3588,6 @@
                 "node": ">=0.4.0"
             }
         },
-        "node_modules/delegates": {
-            "version": "1.0.0",
-            "resolved": "https://registry.npmjs.org/delegates/-/delegates-1.0.0.tgz",
-            "integrity": "sha512-bd2L678uiWATM6m5Z1VzNCErI3jiGzt6HGY8OVICs40JQq/HALfbyNJmp0UDakEY4pMMaN0Ly5om/B1VI/+xfQ==",
-            "optional": true
-        },
         "node_modules/depd": {
             "version": "2.0.0",
             "resolved": "https://registry.npmjs.org/depd/-/depd-2.0.0.tgz",
@@ -3720,18 +3640,6 @@
                 "npm": "1.2.8000 || >= 1.4.16"
             }
         },
-        "node_modules/detect-libc": {
-            "version": "1.0.3",
-            "resolved": "https://registry.npmjs.org/detect-libc/-/detect-libc-1.0.3.tgz",
-            "integrity": "sha512-pGjwhsmsp4kL2RTz08wcOlGN83otlqHeD/Z5T8GXZB+/YcpQ/dgo+lbU8ZsGxV0HIvqqxo9l7mqYwyYMD9bKDg==",
-            "optional": true,
-            "bin": {
-                "detect-libc": "bin/detect-libc.js"
-            },
-            "engines": {
-                "node": ">=0.10"
-            }
-        },
         "node_modules/detective": {
             "version": "5.2.1",
             "resolved": "https://registry.npmjs.org/detective/-/detective-5.2.1.tgz",
@@ -5302,70 +5210,6 @@
                 "url": "https://github.com/sponsors/ljharb"
             }
         },
-        "node_modules/gauge": {
-            "version": "2.7.4",
-            "resolved": "https://registry.npmjs.org/gauge/-/gauge-2.7.4.tgz",
-            "integrity": "sha512-14x4kjc6lkD3ltw589k0NrPD6cCNTD6CWoVUNpB85+DrtONoZn+Rug6xZU5RvSC4+TZPxA5AnBibQYAvZn41Hg==",
-            "deprecated": "This package is no longer supported.",
-            "optional": true,
-            "dependencies": {
-                "aproba": "^1.0.3",
-                "console-control-strings": "^1.0.0",
-                "has-unicode": "^2.0.0",
-                "object-assign": "^4.1.0",
-                "signal-exit": "^3.0.0",
-                "string-width": "^1.0.1",
-                "strip-ansi": "^3.0.1",
-                "wide-align": "^1.1.0"
-            }
-        },
-        "node_modules/gauge/node_modules/ansi-regex": {
-            "version": "2.1.1",
-            "resolved": "https://registry.npmjs.org/ansi-regex/-/ansi-regex-2.1.1.tgz",
-            "integrity": "sha512-TIGnTpdo+E3+pCyAluZvtED5p5wCqLdezCyhPZzKPcxvFplEt4i+W7OONCKgeZFT3+y5NZZfOOS/Bdcanm1MYA==",
-            "optional": true,
-            "engines": {
-                "node": ">=0.10.0"
-            }
-        },
-        "node_modules/gauge/node_modules/is-fullwidth-code-point": {
-            "version": "1.0.0",
-            "resolved": "https://registry.npmjs.org/is-fullwidth-code-point/-/is-fullwidth-code-point-1.0.0.tgz",
-            "integrity": "sha512-1pqUqRjkhPJ9miNq9SwMfdvi6lBJcd6eFxvfaivQhaH3SgisfiuudvFntdKOmxuee/77l+FPjKrQjWvmPjWrRw==",
-            "optional": true,
-            "dependencies": {
-                "number-is-nan": "^1.0.0"
-            },
-            "engines": {
-                "node": ">=0.10.0"
-            }
-        },
-        "node_modules/gauge/node_modules/string-width": {
-            "version": "1.0.2",
-            "resolved": "https://registry.npmjs.org/string-width/-/string-width-1.0.2.tgz",
-            "integrity": "sha512-0XsVpQLnVCXHJfyEs8tC0zpTVIr5PKKsQtkT29IwupnPTjtPmQ3xT/4yCREF9hYkV/3M3kzcUTSAZT6a6h81tw==",
-            "optional": true,
-            "dependencies": {
-                "code-point-at": "^1.0.0",
-                "is-fullwidth-code-point": "^1.0.0",
-                "strip-ansi": "^3.0.0"
-            },
-            "engines": {
-                "node": ">=0.10.0"
-            }
-        },
-        "node_modules/gauge/node_modules/strip-ansi": {
-            "version": "3.0.1",
-            "resolved": "https://registry.npmjs.org/strip-ansi/-/strip-ansi-3.0.1.tgz",
-            "integrity": "sha512-VhumSSbBqDTP8p2ZLKj40UjBCV4+v8bUSEpUb4KjRgWk9pbqGF4REFj6KEagidb2f/M6AzC0EmFyDNGaw9OCzg==",
-            "optional": true,
-            "dependencies": {
-                "ansi-regex": "^2.0.0"
-            },
-            "engines": {
-                "node": ">=0.10.0"
-            }
-        },
         "node_modules/get-assigned-identifiers": {
             "version": "1.2.0",
             "resolved": "https://registry.npmjs.org/get-assigned-identifiers/-/get-assigned-identifiers-1.2.0.tgz",
@@ -5693,12 +5537,6 @@
                 "url": "https://github.com/sponsors/ljharb"
             }
         },
-        "node_modules/has-unicode": {
-            "version": "2.0.1",
-            "resolved": "https://registry.npmjs.org/has-unicode/-/has-unicode-2.0.1.tgz",
-            "integrity": "sha512-8Rf9Y83NBReMnx0gFzA8JImQACstCYWUplepDa9xprwwtmgEZUF0h/i5xSA625zB/I37EtrswSST6OXxwaaIJQ==",
-            "optional": true
-        },
         "node_modules/hash-base": {
             "version": "3.0.4",
             "resolved": "https://registry.npmjs.org/hash-base/-/hash-base-3.0.4.tgz",
@@ -6819,18 +6657,6 @@
                 "node": ">=6"
             }
         },
-        "node_modules/mimic-response": {
-            "version": "2.1.0",
-            "resolved": "https://registry.npmjs.org/mimic-response/-/mimic-response-2.1.0.tgz",
-            "integrity": "sha512-wXqjST+SLt7R009ySCglWBCFpjUygmCIfD790/kVbiGmUgfYGuB14PiTd5DwVxSV4NcYHjzMkoj5LjQZwTQLEA==",
-            "optional": true,
-            "engines": {
-                "node": ">=8"
-            },
-            "funding": {
-                "url": "https://github.com/sponsors/sindresorhus"
-            }
-        },
         "node_modules/minimalistic-assert": {
             "version": "1.0.1",
             "resolved": "https://registry.npmjs.org/minimalistic-assert/-/minimalistic-assert-1.0.1.tgz",
@@ -7296,12 +7122,6 @@
                 "multicast-dns": "cli.js"
             }
         },
-        "node_modules/nan": {
-            "version": "2.14.2",
-            "resolved": "https://registry.npmjs.org/nan/-/nan-2.14.2.tgz",
-            "integrity": "sha512-M2ufzIiINKCuDfBSAUr1vWQ+vuVcA9kqx8JJUsbQi6yf1uGRyb7HfpdfUr5qLXf3B/t8dPvcjhKMmlfnP47EzQ==",
-            "optional": true
-        },
         "node_modules/nanoid": {
             "version": "3.3.1",
             "resolved": "https://registry.npmjs.org/nanoid/-/nanoid-3.3.1.tgz",
@@ -7365,39 +7185,6 @@
                 "@sinonjs/commons": "^3.0.0"
             }
         },
-        "node_modules/node-abi": {
-            "version": "2.30.1",
-            "resolved": "https://registry.npmjs.org/node-abi/-/node-abi-2.30.1.tgz",
-            "integrity": "sha512-/2D0wOQPgaUWzVSVgRMx+trKJRC2UG4SUc4oCJoXx9Uxjtp0Vy3/kt7zcbxHF8+Z/pK3UloLWzBISg72brfy1w==",
-            "optional": true,
-            "dependencies": {
-                "semver": "^5.4.1"
-            }
-        },
-        "node_modules/node-abi/node_modules/semver": {
-            "version": "5.7.2",
-            "resolved": "https://registry.npmjs.org/semver/-/semver-5.7.2.tgz",
-            "integrity": "sha512-cBznnQ9KjJqU67B52RMC65CMarK2600WFnbkcaiwWq3xy/5haFJlshgnpjovMVJ+Hff49d8GEn0b87C5pDQ10g==",
-            "optional": true,
-            "bin": {
-                "semver": "bin/semver"
-            }
-        },
-        "node_modules/node-aead-crypto": {
-            "version": "2.2.2",
-            "resolved": "https://registry.npmjs.org/node-aead-crypto/-/node-aead-crypto-2.2.2.tgz",
-            "integrity": "sha512-EtCLL1FmVjj2GlBNcLRn75ea+y6yGuEdoTpqc9zIiRkIKk1ucTsOPoKaHYYxHfMAXWqHu2Dw8a44VSSKz45UTg==",
-            "hasInstallScript": true,
-            "optional": true,
-            "dependencies": {
-                "bindings": "^1.5.0",
-                "nan": "2.14.x",
-                "prebuild-install": "^6.1.3"
-            },
-            "engines": {
-                "node": ">4"
-            }
-        },
         "node_modules/node-coap-client": {
             "version": "1.0.8",
             "resolved": "https://registry.npmjs.org/node-coap-client/-/node-coap-client-1.0.8.tgz",
@@ -8632,19 +8419,6 @@
                 "node": ">=8"
             }
         },
-        "node_modules/npmlog": {
-            "version": "4.1.2",
-            "resolved": "https://registry.npmjs.org/npmlog/-/npmlog-4.1.2.tgz",
-            "integrity": "sha512-2uUqazuKlTaSI/dC8AzicUck7+IrEaOnN/e0jd3Xtt1KcGpwx30v50mL7oPyr/h9bL3E4aZccVwpwP+5W9Vjkg==",
-            "deprecated": "This package is no longer supported.",
-            "optional": true,
-            "dependencies": {
-                "are-we-there-yet": "~1.1.2",
-                "console-control-strings": "~1.1.0",
-                "gauge": "~2.7.3",
-                "set-blocking": "~2.0.0"
-            }
-        },
         "node_modules/number-allocator": {
             "version": "1.0.14",
             "resolved": "https://registry.npmjs.org/number-allocator/-/number-allocator-1.0.14.tgz",
@@ -8654,24 +8428,6 @@
                 "js-sdsl": "4.3.0"
             }
         },
-        "node_modules/number-is-nan": {
-            "version": "1.0.1",
-            "resolved": "https://registry.npmjs.org/number-is-nan/-/number-is-nan-1.0.1.tgz",
-            "integrity": "sha512-4jbtZXNAsfZbAHiiqjLPBiCl16dES1zI4Hpzzxw61Tk+loF+sBDBKx1ICKKKwIqQ7M0mFn1TmkN7euSncWgHiQ==",
-            "optional": true,
-            "engines": {
-                "node": ">=0.10.0"
-            }
-        },
-        "node_modules/object-assign": {
-            "version": "4.1.1",
-            "resolved": "https://registry.npmjs.org/object-assign/-/object-assign-4.1.1.tgz",
-            "integrity": "sha512-rJgTQnkUnH1sFw8yT6VSU3zD3sWmu6sZhIseY8VX+GRu3P6F7Fu+JNDoXfklElbLJSnc3FUQHVe4cU5hj+BcUg==",
-            "optional": true,
-            "engines": {
-                "node": ">=0.10.0"
-            }
-        },
         "node_modules/object-inspect": {
             "version": "1.13.2",
             "resolved": "https://registry.npmjs.org/object-inspect/-/object-inspect-1.13.2.tgz",
@@ -9152,33 +8908,6 @@
                 "node": ">= 0.4"
             }
         },
-        "node_modules/prebuild-install": {
-            "version": "6.1.4",
-            "resolved": "https://registry.npmjs.org/prebuild-install/-/prebuild-install-6.1.4.tgz",
-            "integrity": "sha512-Z4vpywnK1lBg+zdPCVCsKq0xO66eEV9rWo2zrROGGiRS4JtueBOdlB1FnY8lcy7JsUud/Q3ijUxyWN26Ika0vQ==",
-            "optional": true,
-            "dependencies": {
-                "detect-libc": "^1.0.3",
-                "expand-template": "^2.0.3",
-                "github-from-package": "0.0.0",
-                "minimist": "^1.2.3",
-                "mkdirp-classic": "^0.5.3",
-                "napi-build-utils": "^1.0.1",
-                "node-abi": "^2.21.0",
-                "npmlog": "^4.0.1",
-                "pump": "^3.0.0",
-                "rc": "^1.2.7",
-                "simple-get": "^3.0.3",
-                "tar-fs": "^2.0.0",
-                "tunnel-agent": "^0.6.0"
-            },
-            "bin": {
-                "prebuild-install": "bin.js"
-            },
-            "engines": {
-                "node": ">=6"
-            }
-        },
         "node_modules/precond": {
             "version": "0.2.3",
             "resolved": "https://registry.npmjs.org/precond/-/precond-0.2.3.tgz",
@@ -10049,12 +9778,6 @@
                 "ts-expect": "^1.1.0"
             }
         },
-        "node_modules/set-blocking": {
-            "version": "2.0.0",
-            "resolved": "https://registry.npmjs.org/set-blocking/-/set-blocking-2.0.0.tgz",
-            "integrity": "sha512-KiKBS8AnWGEyLzofFfmvKwpdPzqiy16LvQfK3yv/fVH7Bj13/wl3JSR1J+rfgRE9q7xUJK4qvgS8raSOeLUehw==",
-            "optional": true
-        },
         "node_modules/set-function-length": {
             "version": "1.2.2",
             "resolved": "https://registry.npmjs.org/set-function-length/-/set-function-length-1.2.2.tgz",
@@ -10245,17 +9968,6 @@
                 }
             ]
         },
-        "node_modules/simple-get": {
-            "version": "3.1.1",
-            "resolved": "https://registry.npmjs.org/simple-get/-/simple-get-3.1.1.tgz",
-            "integrity": "sha512-CQ5LTKGfCpvE1K0n2us+kuMPbk/q0EKl82s4aheV9oXjFEz6W/Y7oQFVJuU6QG77hRT4Ghb5RURteF5vnWjupA==",
-            "optional": true,
-            "dependencies": {
-                "decompress-response": "^4.2.0",
-                "once": "^1.3.1",
-                "simple-concat": "^1.0.0"
-            }
-        },
         "node_modules/sinon": {
             "version": "11.1.1",
             "resolved": "https://registry.npmjs.org/sinon/-/sinon-11.1.1.tgz",
@@ -11784,15 +11496,6 @@
                 "url": "https://github.com/sponsors/ljharb"
             }
         },
-        "node_modules/wide-align": {
-            "version": "1.1.5",
-            "resolved": "https://registry.npmjs.org/wide-align/-/wide-align-1.1.5.tgz",
-            "integrity": "sha512-eDMORYaPNZ4sQIuuYPDHdQvf4gyCF9rEEV/yPxGfwPkRodwEgiMUUXTx/dex+Me0wxx53S+NgUHaP7y3MGlDmg==",
-            "optional": true,
-            "dependencies": {
-                "string-width": "^1.0.2 || 2 || 3 || 4"
-            }
-        },
         "node_modules/word-wrap": {
             "version": "1.2.5",
             "resolved": "https://registry.npmjs.org/word-wrap/-/word-wrap-1.2.5.tgz",

```
</details>

**Comment by JKRhb** [comment]
Hmm, okay, but even with a different `package-lock.json`, the test still fails under Node 18 and Windows. Not sure what is causing the issue here :/

**Comment by JKRhb** [comment]
It seems like this problem is related to Windows in general, as the tests also fail under Node 20 there (see https://github.com/JKRhb/thingweb.node-wot/actions/runs/9872206163/job/27261828401?pr=10). Very strange and annoying :/

**Comment by danielpeintner** [comment]
I agree, it is strange and VERY annoying 😒

**Comment by danielpeintner** [comment]
Note: in https://github.com/eclipse-thingweb/node-wot/pull/1274 we had issues with Node.js 22 on Windows ... but maybe the problems are _similar_ !?

**Comment by JKRhb** [comment]
> Note: in #1274 we had issues with Node.js 22 on Windows ... but maybe the problems are _similar_ !?

Oh, yeah, that actually seems likely now :/ I will include Node 22 on Windows in my investigations as well

**Comment by JKRhb** [comment]
> > Note: in #1274 we had issues with Node.js 22 on Windows ... but maybe the problems are similar !?
> 
> There I concluded the the problem was definitely caused in the mbus package, I'm not sure this is the case here.

Thank you! You are right, that seems to actually be the culprit here... I think I got a lot close to the problem by now, but for some reason the test still terminates with a non-zero exit code... Will do some more investigation.

**Comment by JKRhb** [comment]
Hmm, my suggestion for now would skipping the tests for the mbus binding under Windows and investigate the problems separately to be able to move on with the general development.

What is very weird is that the mbus tests seem to succeed, but the exit code is still 1 🤔

**Comment by JKRhb** [comment]
I've now created #1302 as a workaround that would supersede this PR and #1274.

**Pull request closed without merging**